### PR TITLE
variant optional attributes

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

**PR intended to be tested with API branch: variant-optional-attributes

### Screenshots

Screenshot 2020-09-07 at 11 40 32
Screenshot 2020-09-07 at 11 41 11
Screenshot 2020-09-07 at 11 40 55

updated styles:
Screenshot 2020-09-21 at 12 19 11

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://variant-optional-attributes.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [CU-7jw8ef Dashboard optional variant attributes changes](https://app.clickup.com/t/7jw8ef)

File: src/products/components/ProductVariantCreatorPage/ProductVariantChooseAttributes.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      <ResponsiveTable>
+        <MuiTableHead>
+          <TableRow>
+            {(attributes === undefined || attributes.length > 0) && (
+              <TableCell padding="checkbox">
+                <Checkbox
+                  indeterminate={
+                    attributes && attributes.length > selected && selected > 0
+                  }
+                  checked={selected === 0 ? false : true}
```
Comment (orzechdev) [inline src/products/components/ProductVariantCreatorPage/ProductVariantChooseAttributes.tsx:52]:
```suggestion
                  checked={selected !== 0}
```

**Review by orzechdev — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Comment by maarcingebala** [comment]
Please don't merge this one yet. The feature was not fully refined and we need to check if this can be merged in this shape.

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
Screenshot 2020-09-10 at 13 47 42
Screenshot 2020-09-10 at 13 47 56

Doesn't look the same

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
A few things are missing or need fixing:
- [ ] variant creator has "choose attributes" section, but it doesn't have corresponding story
- [ ] interactive story does not work as it's supposed to; if I select "author" and "color", in second view I get to choose "box size" and "coffee genre" 
- [ ] variant creator should not be able to proceed if no attributes were chosen
- [ ] missing tests (`createVariants.test.ts`, `reducer.test.ts`)
- [ ] styles exploded Screenshot 2020-09-21 at 13 26 20

**Comment by krzysztofwolski** [comment]
Blocked #6059

**Comment by maarcingebala** [comment]
Closing this for now, we need to rethink the requirements for this feature and get back to it in the future.

**Pull request closed without merging**